### PR TITLE
ENH: sample features using optional limit keyword argument

### PR DESCRIPTION
**Pull request by GavinHuttley**
[NEW] on all annotation_db classes, added a new optional
    limit argument to both the get_features_matching and
    get_records_matching methods. The argument applies
    across all tables.

[NEW] added optional limit argument on sequence collection
    and on sequence classes to the get_features() methods

[NEW] added docs too

## Summary by Sourcery

Add an optional limit parameter to feature and record retrieval APIs to cap the number of results returned across annotation databases, sequences, and alignments.

New Features:
- Introduce an optional limit argument to annotation_db get_features_matching and get_records_matching methods to constrain the total number of results across all tables.
- Add an optional limit argument to Sequence, SequenceCollection, and Alignment get_features methods to limit the number of features yielded.

Enhancements:
- Plumb result limiting through the SQL selection layer so database queries can be constrained directly rather than post-filtered.

Documentation:
- Document the new limit parameter and its semantics in the annotation database and feature usage guides.

Tests:
- Add tests covering valid, None, and invalid limit values for annotation_db feature and record queries, including multi-table scenarios.
- Add tests for limit handling in sequence, sequence collection, and alignment get_features, including alignment-only and sequence-only feature queries.

### CODE REVIEW

File: tests/test_core/test_annotation_db.py [hunk 1]
```diff
@@ -757,6 +757,145 @@ def test_get_features_matching_start_stop_seqview(DATA_DIR, seq):
...
+    got = list(
+        populated_basic_db.get_features_matching(
+            biotype="exon",
+            start=0,
+            stop=25,
+            allow_partial=True,
+            limit=2,
+        )
+    )
+    assert len(got) == 2
```
[the remark this answers is not in the document]
Comment (GavinHuttley) [reply tests/test_core/test_annotation_db.py:818]: already done

File: src/cogent3/core/sequence.py [hunk 2]
```diff
@@ -1129,6 +1129,7 @@ def get_features(
         start: int | None = None,
         stop: int | None = None,
         allow_partial: bool = False,
+        limit: int | None = None,
         **kwargs: Any,
     ) -> Iterator[Feature[Sequence]]:
         """yields Feature instances
```
[the remark this answers is not in the document]
Comment (GavinHuttley) [reply src/cogent3/core/sequence.py:1135]: The responsibility for deciding a valid value for limit belongs to the class that employs it.
